### PR TITLE
Update mqtt-publisher.py fixes #2431

### LINKED ISSUE #2431: mqtt-publisher user hal component is not sending keys
The mqtt-publisher is connecting to mqtt broker but I cannot get the mqtt-publisher component to send any key values.

pip install paho-mqtt

In HAL file:
loadusr -W mqtt-publisher --mqtt-broker=[MQTT]BROKER --mqtt-user=[MQTT]USERNAME --mqtt-password=[MQTT]PASSWORD keys=Xpos,Ypos,Zpos,halui.machine.is-on,halui.mode.is-auto,halui.mode.is-manual,halui.mode.is-mdi,halui.mode.is-teleop,halui.program.is-running

## This is what I expected to happen:
In MQTT Explorer under devices/linuxcnc/machine
{
  "mqtt-publisher.period": 10
  "Xpos": 1234
  "Ypos": 1234
  ...
}

## This is what happened instead:
MQTT Explorer devices/linuxcnc/machine
{
  "mqtt-publisher.period": 10
}

## Information about my hardware and software:

Machine: VBox Debian GNU/Linux 11 (bullseye)

Config: configs/sim/axis/axis_mm

LinuxCNCDEV 5.10.0-21-amd64 #1 SMP Debian 5.10.162-1 (2023-01-21) x86_64 GNU/Linux

Git version:
v2.9.0-pre1-673-gffb4e43998

### PR DESCRIPTION
**Pull request by mrbubble62**
Fix for missing keys data

### CODE REVIEW

File: src/hal/user_comps/mqtt-publisher.py [hunk 1]
```diff
@@ -84,9 +84,18 @@ def create_pins(self, keys):
...
         "Run the endless loop fetching pin data and sending it to MQTT."
         running = True
+        missing = {}
         while hal.component_exists('halui') and running:
             data = {}
             if self.hal['enable']:
+                for key in self.keys:
+                    try:
+                         data[key] = hal.get_value(key)
+                    except RuntimeError as e:
```
Comment (petterreinholdtsen) [inline src/hal/user_comps/mqtt-publisher.py:94]: Is RuntimeError really the correct exception from get_value(key)?  It seem overly broad.

**Comment by petterreinholdtsen** [comment]
The code look like it should work.  Just a minor question about the exception handling that should be looked at before it is merged.

File: src/hal/user_comps/mqtt-publisher.py (hunk 1 above)
Comment (mrbubble62) [reply src/hal/user_comps/mqtt-publisher.py:94]: Fair comment, AttributeError seems to work just fine.

**Comment by mrbubble62** [comment]
From the build test errors seems like RuntimeError was appropriate..

```
Traceback (most recent call last):
          File "/usr/bin/mqtt-publisher", line 159, in <module>
            LinuxCNC2MQTT.main()
          File "/usr/bin/mqtt-publisher", line 156, in main
            h.update_mqtt()
          File "/usr/bin/mqtt-publisher", line 93, in update_mqtt
            data[key] = hal.get_value(key)
        RuntimeError: Can't set value: pin / param halui.axis.a.pos-feedback not found
        link (updating variable file): No such file or directory
        Traceback (most recent call last):
          File "./display.py", line 42, in <module>
            lastpublish = hal.get_value(name)
        RuntimeError: Can't set value: pin / param mqtt-publisher.lastpublish not found
```

File: src/hal/user_comps/mqtt-publisher.py (hunk 1 above)
Comment (mrbubble62) [reply src/hal/user_comps/mqtt-publisher.py:94]: was testing wrong code, AttributeError does not catch the error thrown.

**Review by petterreinholdtsen — APPROVED** [review]
Look good to me.

**Pull request merged**